client/state/themes/selectors/is-theme-retired.js

```diff
@@ -0,0 +1,15 @@
+import { getTheme } from 'calypso/state/themes/selectors/get-theme';
+import { isRetired } from 'calypso/state/themes/utils';
+
+import 'calypso/state/themes/init';
+
+/**
+ * Whether a WPCOM theme given by its ID is retired.
+ *
+ * @param  {Object} state   Global state tree
+ * @param  {string} themeId Theme ID
+ * @returns {boolean}        True if the theme is retired
+ */
+export function isThemeRetired( state, themeId ) {
+	return isRetired( getTheme( state, 'wpcom', themeId ) );
```

Comment:
The theme might not be fetched. So, one option is to get theme from wporg as well and add the slug to `RETIRED_THEME_SLUGS_SET`

Reply:
The theme was null when I went to the Theme Showcase and searched “Maxwell”. Or maybe you can head to the search result page directly 🤔

Reply:
@arthur791004 with D113775-code merged, the query will no longer return `null`. Could you test again? Thanks!

Reply:
We have to pass `retired=true` to themes endpoint to get the result

Reply:
Oops, I forgot to commit 😅